test/integration/etcd/data.go

```diff
@@ -269,6 +269,17 @@ func GetEtcdStorageDataForNamespace(namespace string) map[schema.GroupVersionRes
 		},
 		// --
 
+		// k8s.io/kubernetes/pkg/apis/storage/v1alpha1
+		gvr("storage.k8s.io", "v1alpha1", "csistoragecapacities"): {
```

Comment:
it is surprising this was not required before. it looks like the API registration was contingent on `if utilfeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity)`, which is unusual. Enabling the alpha API with `--runtime-config=storage.k8s.io/v1alpha1=true` is typically all that is required. Adding the feature gate check meant that this test didn't exercise storage path and version testing

Reply:
> it looks like the API registration was contingent on if utilfeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity), which is unusual.

What if the v1alpha1 API group had had two different APIs which are under different feature gates? From that perspective it made sense to me to toggle the API individually based on the feature gate.

What about the REST storage provider? There basically all API types get added depending on the feature gates, for example:
https://github.com/kubernetes/kubernetes/blob/603a2aa8a99869dc51ea1be2ce6e6ca10c23e55f/pkg/registry/storage/rest/storage_storage.go#L98-L105

Is that normal or unusual?

But I agree, this did have the undesired effect that several tests simply didn't run for the CSIStorageCapacity object before.

Reply:
> What if the v1alpha1 API group had had two different APIs which are under different feature gates? From that perspective it made sense to me to toggle the API individually based on the feature gate.
> 
> What about the REST storage provider? There basically all API types get added depending on the feature gates, for example:
> ...
> Is that normal or unusual?

That's unusual. Opting into enabling an alpha API version typically gets you everything in the version. Disabling beta APIs is typically done at the version level.